app/controllers/service_provider_controller.rb

```diff
@@ -0,0 +1,13 @@
+class ServiceProviderController < ApplicationController
+  protect_from_forgery with: :null_session
+
+  def update
+    if FeatureManagement.use_dashboard_service_providers?
+      ServiceProviderUpdater.new.run
+      SecureHeadersWhitelister.new.run
```

Comment:
Why does this need to be run a second time? This gets called at initialization by `config/initializers/secure_headers.rb`.

Also, in order for this controller to take advantage of this SecureHeader configuration override, it must declare a before_action that tells it to use the override, like here: 
https://github.com/18F/identity-idp/blob/master/app/controllers/saml_idp_controller.rb#L13

https://github.com/18F/identity-idp/blob/master/app/controllers/saml_idp_controller.rb#L60-L62

Would it make more sense to move the `SERVICE_PROVIDERS` assignment out of its own initializer and into the `SecureHeadersWhitelister` class, such that it sets the constant to the proper value based on whether or not `FeatureManagement.use_dashboard_service_providers?` is true?